File: markopiApp/backend/Back-End-Reengineering-Markopi/app/Models/Iklan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Iklan extends Model
{
    use HasFactory;

    protected $table = 'iklan';

    protected $primaryKey = 'id_iklan';

    public $timestamps = false; 

    protected $fillable = [
        'judul_iklan',
        'deskripsi_iklan',
        'gambar',
        'link',
    ];
}

File: markopiApp/markopi/lib/models/Iklan.dart
class Iklan {
  final int idIklan;
  final String judulIklan;
  final String deskripsiIklan;
  final String gambar;
  final String link;

  Iklan({
    required this.idIklan,
    required this.judulIklan,
    required this.deskripsiIklan,
    required this.gambar,
    required this.link,
  });

  factory Iklan.fromJson(Map<String, dynamic> json) {
    return Iklan(
      idIklan: json['id_iklan'],
      judulIklan: json['judul_iklan'],
      deskripsiIklan: json['deskripsi_iklan'],
      gambar: json['gambar'],
      link: json['link'],
    );
  }
}
